MOOSE Makefiles fail cryptically when checkout path contains spaces

### Bug Description

When the MOOSE checkout path contains whitespace, the Make-based build can fail before reporting a useful MOOSE-level diagnostic. For example, a checkout under `/Users/saqif/Documents/OpenSees Project/MOOSE` failed immediately with:

```
make: *** /Users/saqif/Documents: Is a directory. Stop.
```

The failure occurs because paths such as `$(FRAMEWORK_DIR)/build.mk` are included without whitespace-safe handling, so Make splits the include path before the framework makefiles can run any checks.

### Steps to Reproduce

1. Place a MOOSE checkout under a directory containing a space.
2. Run `make -C test -n` or another standard MOOSE Makefile entrypoint.

### Impact

This is an onboarding and support issue. Users get a low-level Make error that points at only the first path component, instead of a clear message explaining that MOOSE Makefiles do not support whitespace in `MOOSE_DIR` or `FRAMEWORK_DIR`.

--- a/examples/Makefile
+++ b/examples/Makefile
@@ -7,8 +7,14 @@
 # FRAMEWORK_DIR - Location of the MOOSE framework
 #
 ###############################################################################
-MOOSE_DIR          ?= $(shell dirname `pwd`)
+MOOSE_DIR          ?= $(shell dirname "$$PWD")
 FRAMEWORK_DIR      ?= $(MOOSE_DIR)/framework
+ifneq ($(words $(MOOSE_DIR)),1)
+  $(error MOOSE_DIR contains whitespace: '$(MOOSE_DIR)'. MOOSE Makefiles do not support paths with whitespace; move the checkout or set MOOSE_DIR to a path without whitespace)
+endif
+ifneq ($(words $(FRAMEWORK_DIR)),1)
+  $(error FRAMEWORK_DIR contains whitespace: '$(FRAMEWORK_DIR)'. MOOSE Makefiles do not support paths with whitespace; move the checkout or set FRAMEWORK_DIR to a path without whitespace)
+endif
 ADDITIONAL_CPPFLAGS += -Wall -Wextra
 ###############################################################################
 

--- a/framework/Makefile
+++ b/framework/Makefile
@@ -7,8 +7,14 @@
 # FRAMEWORK_DIR - Location of the MOOSE framework
 #
 ###############################################################################
-MOOSE_DIR          ?= $(shell dirname `pwd`)
+MOOSE_DIR          ?= $(shell dirname "$$PWD")
 FRAMEWORK_DIR      ?= $(MOOSE_DIR)/framework
+ifneq ($(words $(MOOSE_DIR)),1)
+  $(error MOOSE_DIR contains whitespace: '$(MOOSE_DIR)'. MOOSE Makefiles do not support paths with whitespace; move the checkout or set MOOSE_DIR to a path without whitespace)
+endif
+ifneq ($(words $(FRAMEWORK_DIR)),1)
+  $(error FRAMEWORK_DIR contains whitespace: '$(FRAMEWORK_DIR)'. MOOSE Makefiles do not support paths with whitespace; move the checkout or set FRAMEWORK_DIR to a path without whitespace)
+endif
 ADDITIONAL_CPPFLAGS += -Wall -Wextra
 ###############################################################################
 

--- a/modules/Makefile
+++ b/modules/Makefile
@@ -7,8 +7,14 @@
 # FRAMEWORK_DIR - Location of the MOOSE framework
 #
 ###############################################################################
-MOOSE_DIR          ?= $(shell dirname `pwd`)
+MOOSE_DIR          ?= $(shell dirname "$$PWD")
 FRAMEWORK_DIR      ?= $(MOOSE_DIR)/framework
+ifneq ($(words $(MOOSE_DIR)),1)
+  $(error MOOSE_DIR contains whitespace: '$(MOOSE_DIR)'. MOOSE Makefiles do not support paths with whitespace; move the checkout or set MOOSE_DIR to a path without whitespace)
+endif
+ifneq ($(words $(FRAMEWORK_DIR)),1)
+  $(error FRAMEWORK_DIR contains whitespace: '$(FRAMEWORK_DIR)'. MOOSE Makefiles do not support paths with whitespace; move the checkout or set FRAMEWORK_DIR to a path without whitespace)
+endif
 ADDITIONAL_CPPFLAGS += -Wall -Wextra
 ###############################################################################
 

--- a/test/Makefile
+++ b/test/Makefile
@@ -7,8 +7,14 @@
 # FRAMEWORK_DIR - Location of the MOOSE framework
 #
 ###############################################################################
-MOOSE_DIR          ?= $(shell dirname `pwd`)
+MOOSE_DIR          ?= $(shell dirname "$$PWD")
 FRAMEWORK_DIR      ?= $(MOOSE_DIR)/framework
+ifneq ($(words $(MOOSE_DIR)),1)
+  $(error MOOSE_DIR contains whitespace: '$(MOOSE_DIR)'. MOOSE Makefiles do not support paths with whitespace; move the checkout or set MOOSE_DIR to a path without whitespace)
+endif
+ifneq ($(words $(FRAMEWORK_DIR)),1)
+  $(error FRAMEWORK_DIR contains whitespace: '$(FRAMEWORK_DIR)'. MOOSE Makefiles do not support paths with whitespace; move the checkout or set FRAMEWORK_DIR to a path without whitespace)
+endif
 ADDITIONAL_CPPFLAGS += -Wall -Wextra
 ###############################################################################
 

--- a/unit/Makefile
+++ b/unit/Makefile
@@ -10,8 +10,14 @@
 # FRAMEWORK_DIR - Location of the MOOSE framework
 #
 ###############################################################################
-MOOSE_DIR          ?= $(shell dirname `pwd`)
+MOOSE_DIR          ?= $(shell dirname "$$PWD")
 FRAMEWORK_DIR      ?= $(MOOSE_DIR)/framework
+ifneq ($(words $(MOOSE_DIR)),1)
+  $(error MOOSE_DIR contains whitespace: '$(MOOSE_DIR)'. MOOSE Makefiles do not support paths with whitespace; move the checkout or set MOOSE_DIR to a path without whitespace)
+endif
+ifneq ($(words $(FRAMEWORK_DIR)),1)
+  $(error FRAMEWORK_DIR contains whitespace: '$(FRAMEWORK_DIR)'. MOOSE Makefiles do not support paths with whitespace; move the checkout or set FRAMEWORK_DIR to a path without whitespace)
+endif
 ADDITIONAL_CPPFLAGS += -Wall -Wextra -DMOOSE_UNIT_TEST
 ###############################################################################
 

